### ISSUE TITLE
Add warning message if model uses `input_ids` that include padding tokens, but no `attention_mask` is provided.

### ISSUE THREAD

**Issue by patrickvonplaten** [huggingface/transformers#16136]
## **First good issue** 

A current error is that a user forwards a batched tensor of `input_ids` that include a padding token, e.g. ```input_ids = torch.tensor([["hello", "this", "is", "a", "long", "string"], ["hello", "<pad>", "<pad>", "<pad>", "<pad>"]]```

In this case, the `attention_mask` should be provided as well. Otherwise the output hidden_states will be incorrectly computed. This is quite a common silent error IMO. 

With @LysandreJik @sgugger, we have decided to not **automatically** create the `attention_mask` that masks out the padding tokens in this case because of the reasons explains here: https://github.com/huggingface/transformers/issues/15479#issuecomment-1066639938 . However as pointed out in https://github.com/huggingface/transformers/issues/15479, we should IMO at least displa a warning since this error happens a lot IMO. 

As a first good issue, one could add such a warning to the BertModel in a first case which would go something like:

```py
if attention_mask is not None and (input_ids == pad_token_id).any():
    logger.warn("display nice warning here....")
```

What do you think @sgugger @LysandreJik ?

**Comment by patrickvonplaten** [comment]
Hmm, think we can be pretty confident that `self.config.pad_token_id` inside the model is the correct padding token. Agree that performance would suffer here a bit. Think putting it in the Trouble shooting guide is a good idea cc @stevhliu

**Comment by stevhliu** [comment]
Yay more content for the troubleshooting guide! I'll work on a PR for this 👍

**Comment by Pawank06** [comment]
Hey, @patrickvonplaten can I work on this issue?

**Comment by patrickvonplaten** [comment]
Sure that'd be great. Just to make sure we don't do duplicated work here - @ydshieh you haven't started on this one yet no?

**Comment by ydshieh** [comment]
Hi, @Pawank06 @patrickvonplaten 

Not really. On Sep. 2022,  I rebased the branch @patrickvonplaten created [add_important_warning_padding_attention_mask]( https://github.com/huggingface/transformers/tree/add_important_warning_padding_attention_mask), but then turned my focus to other tasks.

@Pawank06, maybe you can pull that branch, rebase on the latest main, and continue what @patrickvonplaten has done? Don't hesitate if you need any help ❤️

**Comment by Pawank06** [comment]
@ydshieh @patrickvonplaten Ok can you assign me this issue and  also can you please share me the file path

**Comment by anruijian** [comment]
@ydshieh @Pawank06 Hello, if no one is actively working on this issue, I am willing to take a look and continue the work!

**Comment by ydshieh** [comment]
@anruijian Let's wait a bit for @Pawank06 's response :-) Thank you for expressing the interest 💯

**Comment by anruijian** [comment]
@ydshieh Sure. It seems @Pawank06 removed the assignment.

**Comment by ydshieh** [comment]
I see. @anruijian , you can take a look on [this comment](https://github.com/huggingface/transformers/issues/16136#issuecomment-1416072271), and let me know if you have any question before working on it. Thank you!

**Comment by anruijian** [comment]
@ydshieh I have checked the [add_important_warning_padding_attention_mask](https://github.com/huggingface/transformers/tree/add_important_warning_padding_attention_mask) and would like to confirm my understanding of the current status and next steps before proceeding with my work. As of now, the task has been completed for the Torch version. The next steps involve adding an equivalent warning function to the TensorFlow and Flax versions. More specifically, in [FlaxPreTrainedModel](https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_flax_utils.py#L157), [modeling_flax_bert.py](https://github.com/huggingface/transformers/blob/main/src/transformers/models/bert/modeling_flax_bert.py)and [TFPreTrainedModel](https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_tf_utils.py#L1076), [modeling_tf_bert.py](https://github.com/huggingface/transformers/blob/main/src/transformers/models/bert/modeling_tf_bert.py). Thank you!

**Comment by ydshieh** [comment]
Hi @anruijian . No, the torch part is not finished yet. @patrickvonplaten added a method `warn_if_pad_token_in_input_ids_no_attention_mask` in `src/transformers/modeling_utils.py`, and only used that method in a modeling file `src/transformers/models/bert/modeling_bert.py`.

The goal is to have the same change made in `modeling_bert.py` to other pytorch modeling files in `transformers`, like GPT2, Bart, T5, etc., wherever it makes sense, mostly will be in the places where we have

```python
        elif input_ids is not None:
            input_shape = input_ids.size()
```

**Comment by hackyon** [comment]
@patrickvonplaten @ydshieh It looks like none of the pull requests were committed yet, I'd like to take a stab at this issue if it's ok. Thanks.

**Comment by ydshieh** [comment]
@hackyon Sure!

You can probably continue the work from the branch opened 

https://github.com/huggingface/transformers/pull/21916
